hack/lib/util.sh

```diff
@@ -698,7 +698,9 @@ function kube::util::ensure_dockerized {
 #  SED: The name of the gnu-sed binary
 #
 function kube::util::ensure-gnu-sed {
-  if LANG=C sed --help 2>&1 | grep -q GNU; then
+  # NOTE: the echo below is a workaround to ensure sed is executed before the grep.
+  # see: https://github.com/kubernetes/kubernetes/issues/87251
+  if echo "$(LANG=C sed --help 2>&1)" | grep -q "GNU\|BusyBox"; then
```

Comment:
I strongly doubt that the shell runs the piped commands randomly run out-of-order. The stdout logs that we've seen is probably a result of the buffered output being re-combined incorrectly.

Reply:
Indeed, it sounds incredible, but it is also the only explanation as to why that function would randonly fail.

Reply:
Are we sure that this function fails randomly *on the same system*, and not that the function is failing (seemingly) randomly on different systems that have (or don't have) GNU sed?

Reply:
This is basically what is being executed: https://paste.ubuntu.com/p/2jHTmfCvpP/

In my ubuntu 16.04 (GNU bash, version 5.0.0(1)-release (x86_64-pc-linux-gnu) and my 18.04 (GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu) OSes, it passed for all 200 runs.

While for the ``gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4`` image ("Alpine Linux v3.10") (GNU bash, version 5.0.0(1)-release (x86_64-alpine-linux-musl), the number of failures varies between 19 and 98.